File: MochaDoomCore.Core/w/AidedReadableDoomObject.cs
using System.IO;

namespace w
{
    /**
 * This is for objects that can be read from disk, but cannot
 * self-determine their own.Length for some reason.
 *
 * @author Maes
 */

    public interface AidedReadableDoomObject
    {

        void read(Stream f, int len);
    }
}

File: MochaDoomCore.Core/w/AidedReadableDoomObject.java
namespace w {  

using java.io.DataInputStream;
using java.io.IOException;

/**
 * This is for objects that can be read from disk, but cannot
 * self-determine their own length for some reason.
 *
 * @author Maes
 */

public interface AidedReadableDoomObject
{

    void read(DataInputStream f, int len)  ;
}
